h5py/tests/test_group.py

```diff
@@ -394,25 +394,59 @@ def test_iter_zero(self):
             hfile.close()
 
 class TestTrackOrder(BaseGroup):
-    def populate(self, g):
-        for i in range(100):
-            # Mix group and dataset creation.
-            if i % 10 == 0:
-                g.create_group(str(i))
-            else:
-                g[str(i)] = [i]
-
-    def test_track_order(self):
-        g = self.f.create_group('order', track_order=True)  # creation order
-        self.populate(g)
-        self.assertEqual(list(g),
-                         [str(i) for i in range(100)])
-
-    def test_no_track_order(self):
-        g = self.f.create_group('order', track_order=False)  # name alphanumeric
-        self.populate(g)
-        self.assertEqual(list(g),
-                         sorted([str(i) for i in range(100)]))
+
+    def test_with_track_order(self):
+        grp = self.f.create_group('b', track_order=True)
+        self.f.create_group('a', track_order=True)
+        self.f.create_group('c', track_order=True)
```

Comment:
uhh, the `populate()` method is well designed, but not good for test_group, the existing bug have proved this, it is not a very straightforward way to show track_order feature for group, we should focus on testing root group and sub group, but not mixing group and dataset, so I propose to write it like this:

```python
def populate(self, g, track_order):
    for i in range(9,0,-1):
        if i % 2 == 0:
            self.f.create_group(str(i), track_order)  # create root group
        else:
            g.create_group(str(i), track_order)       # create sub group

@ut.expectedFailure
def test_track_order(self):
    grp = self.f.create_group('0', track_order=True)   # --->True
    self.populate(grp, True)  # --->True
    expected = ['8', '6', '4', '2', '0']
    # this will fail until issue 1577 fixed
    self.assertEqual(list(self.f.keys(), expected))
   
    expected = ['9', '7', '5', '3', '1']
    self.assertEqual(list(grp.keys(), expected))

def test_no_track_order(self):
    grp = self.f.create_group('0', track_order=False)
    self.populate(grp, False)
    expected = ['0', '2', '4', '6', '8']
    self.assertEqual(list(self.f.keys(), expected))
   
    expected = ['1', '3', '5', '7', '9']
    self.assertEqual(list(grp.keys(), expected))

Reply:
@takluyver Ready for your review.